Possible allure-python incompatibility with pytest-flakes/pytest-pycodestyle

[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [x] bug report

#### What is the current behavior?
When trying to run tests with `pytest` and produce report data with `allure-pytest` the tests suite ends with errors for every tested module (tests and application code style checks).

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem

* Prepare the environment and config (see *Other information* and *Environment* section)
* Prepare a simple test in a module (`test_example.py`) in your test directory:
```
import pytest


@pytest.mark.unit
@pytest.mark.asyncio
class TestExample:
    """Fake test to check the test run"""

    async def test_example(self):
        assert True
```
* Run `py.test --codestyle --flakes --alluredir=allure-results`


#### What is the expected behavior?
Tests fail or pass, no errors during the test run.

#### Please tell us about your environment:

- Test framework: pytest@4.2.0
- Allure adaptor: allure-pytest@2.5.5
- Pytest asyncio plugin: pytest-asyncio@0.10.0
- Pytest coverage plugin: pytest-cov@2.6.1
- Pytest flakes plugin: pytest-flakes@4.0.0
- Pytest pycodestyle plugin: pytest-pycodestyle@1.4.0

`pytest.ini`:
```
[pytest]
norecursedirs =
    .eggs
    *.egg
    .ropeproject
    build
testpaths =
    tests
codestyle_ignore = E712 W503 W504 W605
codestyle_exclude =
    migrations
codestyle_max_line_length = 99
flakes-ignore =
    *.py E712
    conftest.py ALL
filterwarnings =
    ignore::DeprecationWarning
    ignore::PendingDeprecationWarning
```

#### Other information 
The stack trace for the errors for the `test_example.py` (all other errors are of the same kind):
```
_____________________________________________________________________________________________________________ ERROR at teardown of pyflakes-check(ignoring E712) _____________________________________________________________________________________________________________

self = <_HookCaller 'pytest_runtest_teardown'>, args = (), kwargs = {'item': <FlakesItem tests/unit/test_example.py>, 'nextitem': <Item tests/unit/test_example.py>}, notincall = set()

    def __call__(self, *args, **kwargs):
        if args:
            raise TypeError("hook calling supports only keyword arguments")
        assert not self.is_historic()
        if self.spec and self.spec.argnames:
            notincall = (
                set(self.spec.argnames) - set(["__multicall__"]) - set(kwargs.keys())
            )
            if notincall:
                warnings.warn(
                    "Argument(s) {} which are declared in the hookspec "
                    "can not be found in this hook call".format(tuple(notincall)),
                    stacklevel=2,
                )
>       return self._hookexec(self, self.get_hookimpls(), kwargs)

../environments/observer_env/lib/python3.7/site-packages/pluggy/hooks.py:284:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x10d67b198>, hook = <_HookCaller 'pytest_runtest_teardown'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/Users/dmitryn/Projects/environments/observer_e...t 0x10ee7f4a8>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10ee899e8>>]
kwargs = {'item': <FlakesItem tests/unit/test_example.py>, 'nextitem': <Item tests/unit/test_example.py>}

    def _hookexec(self, hook, methods, kwargs):
        # called from all hookcaller instances.
        # enable_tracing will set its own wrapping function at self._inner_hookexec
>       return self._inner_hookexec(hook, methods, kwargs)

../environments/observer_env/lib/python3.7/site-packages/pluggy/manager.py:68:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hook = <_HookCaller 'pytest_runtest_teardown'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/Users/dmitryn/Projects/environments/observer_e...t 0x10ee7f4a8>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10ee899e8>>]
kwargs = {'item': <FlakesItem tests/unit/test_example.py>, 'nextitem': <Item tests/unit/test_example.py>}

    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
        methods,
        kwargs,
>       firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
    )

../environments/observer_env/lib/python3.7/site-packages/pluggy/manager.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <allure_pytest.listener.AllureListener object at 0x10ee7f4a8>, item = <FlakesItem tests/unit/test_example.py>

    @pytest.hookimpl(hookwrapper=True)
    def pytest_runtest_teardown(self, item):
        yield
        uuid = self._cache.get(item.nodeid)
        test_result = self.allure_logger.get_test(uuid)
        test_result.labels.extend([Label(name=name, value=value) for name, value in allure_labels(item)])
        test_result.labels.extend([Label(name=LabelType.TAG, value=value) for value in pytest_markers(item)])
>       test_result.labels.extend([Label(name=name, value=value) for name, value in allure_suite_labels(item)])

../environments/observer_env/lib/python3.7/site-packages/allure_pytest/listener.py:104:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

item = <FlakesItem tests/unit/test_example.py>

    def allure_suite_labels(item):
>       head, possibly_clazz, tail = islice(chain(item.nodeid.split('::'), [None]), 3)
E       ValueError: not enough values to unpack (expected 3, got 2)

../environments/observer_env/lib/python3.7/site-packages/allure_pytest/utils.py:110: ValueError
___________________________________________________________________________________________________________________ ERROR at teardown of pycodestyle-check ___________________________________________________________________________________________________________________

self = <_HookCaller 'pytest_runtest_teardown'>, args = (), kwargs = {'item': <Item tests/unit/test_example.py>, 'nextitem': <Function test_example>}, notincall = set()

    def __call__(self, *args, **kwargs):
        if args:
            raise TypeError("hook calling supports only keyword arguments")
        assert not self.is_historic()
        if self.spec and self.spec.argnames:
            notincall = (
                set(self.spec.argnames) - set(["__multicall__"]) - set(kwargs.keys())
            )
            if notincall:
                warnings.warn(
                    "Argument(s) {} which are declared in the hookspec "
                    "can not be found in this hook call".format(tuple(notincall)),
                    stacklevel=2,
                )
>       return self._hookexec(self, self.get_hookimpls(), kwargs)

../environments/observer_env/lib/python3.7/site-packages/pluggy/hooks.py:284:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x10d67b198>, hook = <_HookCaller 'pytest_runtest_teardown'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/Users/dmitryn/Projects/environments/observer_e...t 0x10ee7f4a8>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10ee899e8>>]
kwargs = {'item': <Item tests/unit/test_example.py>, 'nextitem': <Function test_example>}

    def _hookexec(self, hook, methods, kwargs):
        # called from all hookcaller instances.
        # enable_tracing will set its own wrapping function at self._inner_hookexec
>       return self._inner_hookexec(hook, methods, kwargs)

../environments/observer_env/lib/python3.7/site-packages/pluggy/manager.py:68:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hook = <_HookCaller 'pytest_runtest_teardown'>
methods = [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/Users/dmitryn/Projects/environments/observer_e...t 0x10ee7f4a8>>, <HookImpl plugin_name='logging-plugin', plugin=<_pytest.logging.LoggingPlugin object at 0x10ee899e8>>]
kwargs = {'item': <Item tests/unit/test_example.py>, 'nextitem': <Function test_example>}

    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
        methods,
        kwargs,
>       firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
    )

../environments/observer_env/lib/python3.7/site-packages/pluggy/manager.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <allure_pytest.listener.AllureListener object at 0x10ee7f4a8>, item = <Item tests/unit/test_example.py>

    @pytest.hookimpl(hookwrapper=True)
    def pytest_runtest_teardown(self, item):
        yield
        uuid = self._cache.get(item.nodeid)
        test_result = self.allure_logger.get_test(uuid)
        test_result.labels.extend([Label(name=name, value=value) for name, value in allure_labels(item)])
        test_result.labels.extend([Label(name=LabelType.TAG, value=value) for value in pytest_markers(item)])
>       test_result.labels.extend([Label(name=name, value=value) for name, value in allure_suite_labels(item)])

../environments/observer_env/lib/python3.7/site-packages/allure_pytest/listener.py:104:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

item = <Item tests/unit/test_example.py>

    def allure_suite_labels(item):
>       head, possibly_clazz, tail = islice(chain(item.nodeid.split('::'), [None]), 3)
E       ValueError: not enough values to unpack (expected 3, got 2)

../environments/observer_env/lib/python3.7/site-packages/allure_pytest/utils.py:110: ValueError
```
[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions 
. how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

--- a/allure-pytest/src/utils.py
+++ b/allure-pytest/src/utils.py
@@ -107,7 +107,7 @@ def allure_full_name(item):
 
 
 def allure_suite_labels(item):
-    head, possibly_clazz, tail = islice(chain(item.nodeid.split('::'), [None]), 3)
+    head, possibly_clazz, tail = islice(chain(item.nodeid.split('::'), [None], [None]), 3)
     clazz = possibly_clazz if tail else None
     file_name, path = islice(chain(reversed(head.rsplit('/', 1)), [None]), 2)
     module = file_name.split('.')[0]

--- a/allure-pytest/test/integration/pytest_flakes/pytest_flakes_test.py
+++ b/allure-pytest/test/integration/pytest_flakes/pytest_flakes_test.py
@@ -1,14 +1,28 @@
-import pytest
+from hamcrest import assert_that
+from allure_commons_test.report import has_test_case
+from allure_commons_test.result import with_status
 
 
-@pytest.mark.xfail()
-def test_pytest_flakes(allured_testdir, request):
+def test_pytest_flakes(allured_testdir):
     """
+    >>> from os.path import *
     >>> def test_pytest_flakes_example():
     ...     assert True
     """
 
-    allured_testdir.parse_docstring_source(request)
+    allured_testdir.parse_docstring_source()
     allured_testdir.run_with_allure("--flakes")
 
-    assert False, allured_testdir.allure_report
+    assert_that(allured_testdir.allure_report,
+                has_test_case("test_pytest_flakes.py",
+                              with_status("broken")
+                              ),
+
+                )
+
+    assert_that(allured_testdir.allure_report,
+                has_test_case("test_pytest_flakes_example",
+                              with_status("passed")
+                              )
+
+                )

